components/director/internal/domain/labeldef/service.go

```diff
@@ -60,3 +75,45 @@ func (s *service) List(ctx context.Context, tenant string) ([]model.LabelDefinit
 	}
 	return defs, nil
 }
+
+func (s *service) Update(ctx context.Context, def model.LabelDefinition) (model.LabelDefinition, error) {
+	if err := def.ValidateForUpdate(); err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while validating Label Definition")
+	}
+
+	ld, err := s.repo.GetByKey(ctx, def.Tenant, def.Key)
+	if err != nil {
+		return model.LabelDefinition{}, errors.Wrap(err, "while receiving Label Definition")
+	}
+
+	if ld == nil {
+		return model.LabelDefinition{}, fmt.Errorf("definition with %s key doesn't exist", def.Key)
+	}
+
+	ld.Schema = def.Schema
+
+	existingLabels, err := s.labelRepo.ListByKey(ctx, def.Tenant, def.Key)
+
+	validator, err := jsonschema.NewValidatorFromRawSchema(*def.Schema)
```

Comment:
def.Schema can be nil -> then panic will be thrown. But if schema is empty or nil, we don't have to fetch labels from db and validate them.

Reply:
done